src/main/java/jenkins/scm/impl/subversion/SubversionSCMSource.java

```diff
@@ -701,16 +723,29 @@ public SubversionSCM build(@NonNull SCMHead head,
          * The subversion revision.
          */
         private long revision;
+        /**
+         * Indicates whether externals are used. If so, revision is considered non-deterministic.
+         */
+        private boolean usesExternals;
 
-        public SCMRevisionImpl(SCMHead head, long revision) {
+        public SCMRevisionImpl(SCMHead head, long revision, boolean usesExternals) {
             super(head);
             this.revision = revision;
+            this.usesExternals = usesExternals;
         }
 
         public long getRevision() {
             return revision;
         }
 
+        public boolean isDeterministic() {
+            //non-deterministic when using externals, to force Jenkins to recurse into subdirectories
+            if(usesExternals) {
+                return false;
```

Comment:
Nondeterministic revisions are acceptable only for SCMs which inherently cannot support determinism, like CVS I suppose (though even there `-D` timestamps may offer a solution). In the case of Subversion with externals, we would rather need to extend `SCMRevisionImpl` to encode the revision number in each external encountered.